rtt/internal/ChannelBufferElement.hpp

```diff
@@ -129,24 +133,31 @@ namespace RTT { namespace internal {
          */
         virtual void clear()
         {
-	    if(last_sample_p)
-		buffer->Release(last_sample_p);
-	    last_sample_p = 0;
+            if(last_sample_p)
+                buffer->Release(last_sample_p);
+            last_sample_p = 0;
             buffer->clear();
             base::ChannelElement<T>::clear();
         }
 
         virtual bool data_sample(param_t sample)
         {
-            buffer->data_sample(sample);
+            if (!buffer->data_sample(sample)) return false;
```

Comment:
Is it on purpose that the call to the base class is bypassed if `buffer->data_sample(sample)` returns false ? I kind of fail to really see what possible consequences this could have...